### ISSUE TITLE
Hide all tables/collections as default

### ISSUE THREAD

**Issue by felipesabino** [metabase/metabase#2146]
One MongoDB database I am trying to use metabase with has more than 500 collections with tons of metadata and "useless" information (useless for metabase's scope)

Right now what I am doing is going through the DB and executing the query bellow to hide them all, and then enabling just the one or two I actually need.

Is there a better way to have all collections/tables hidden by default or to accomplish the same end result without having to dive into SQL?

```
UPDATE metabase_table
SET visibility_type = 'hidden'
WHERE db_id = <add your db id here, get it at metabase_database table>
```

*  I am using the Google Chome browser.
*  My computer's OS is MacOS.
*  My database is MongoDB.
*  My Metabase version is v0.15.1

**Comment by MrMauricioLeite** [comment]
It would be awesome if there was a button to Hide All in the datamodel admin.

**Comment by mazameli** [comment]
Yeah, I was going to suggest that, @MrMauricioLeite. Good thought. Do you think a Hide All / Show All toggle in the Admin Panel would work for you, @felipesabino?

**Comment by felipesabino** [comment]
@mazameli 

Yes, I guess it would!

It is basically an UI element to execute what I did with the SQL query :smile:

**Comment by mazameli** [comment]
Great. Let me try some layout ideas. I'll post them in here soon.

**Comment by MartinArens** [comment]
+1

**Comment by felipesabino** [comment]
@MartinArens please don't spam other users with `+1` comments, use _Reactions_ instead https://github.com/blog/2119-add-reactions-to-pull-requests-issues-and-comments

**Comment by mazameli** [comment]
I had a couple of free minutes to noodle around with a quick and dirty concept for this. The idea is that rather than having the metadata strength bar next to each table (which is ambiguous and confusing anyway IMHO), we put an eyeball icon to make it easier to show/hide a table without having to click into the table's details. There's also an icon at the top of the column, next the heading that says how many query-able tables there are, which would let you show/hide ALL tables in the current DB or schema/namespace. Thoughts?

![hide all](https://cloud.githubusercontent.com/assets/2223916/15371577/fe6f4926-1cf0-11e6-9ef9-2239780a9efd.png)

![hide all hover](https://cloud.githubusercontent.com/assets/2223916/15371578/fe6f5772-1cf0-11e6-8909-498690f46709.png)

**Comment by kdoh** [comment]
Huge +1 to the functionality. I do think we can probably come up with something a bit nicer for the hidden table icon besides the red strikethrough and maybe bump the contrast on the icon generally, but those are minor details.

**Comment by MrMauricioLeite** [comment]
Great idea, would work fine for us.

**Comment by MrMauricioLeite** [comment]
Guys, no update on this feature? I'm again going through 100+ tables hiding one by one... Takes so much time that I had to come here and see if this was merged.

**Comment by Tall-Paul** [comment]
I've just googled 'metabase hide all tables' and it brought me here, this is something I'd love to see. 

My use case is a magento database which out of the box contains about 400 tables, with less than 20 actually being useful from a reporting point of view.

**Comment by sramazzina** [comment]
Sounds like a great feature. We're experiencing the same problem over an Adempiere database. Would be great to have this feature asap.

**Comment by DrSnap** [comment]
Yes great feature! I'm curious though, what would be the default setting for new tables added to a schema? in any case it would be helpful if our default setting is "Hide until unhidden" that also newly created tables would be hidden at first.

**Comment by mazameli** [comment]
Been meaning for a while to update these designs and clean them up a bit, too. I think this could be a quick and easy UX win.

Hide all button and icons next to hidden tables:

![hide all](https://user-images.githubusercontent.com/2223916/36005055-a5889876-0cea-11e8-8693-0a91d2358156.png)

Hover state:

![hide all button hover](https://user-images.githubusercontent.com/2223916/36005061-aa1a94e8-0cea-11e8-8bc3-062cc08e6d93.png)

Tables hidden:

![all tables hidden](https://user-images.githubusercontent.com/2223916/36005065-aebe5dc2-0cea-11e8-91d3-df602f8b4dd0.png)

The same pattern could apply for schema lists:

![schemas](https://user-images.githubusercontent.com/2223916/36005109-d71e07d6-0cea-11e8-8305-5bbc4c1c411a.png)

We could do the same with the upcoming metadata pages, too:

![metadata page tables](https://user-images.githubusercontent.com/2223916/36005122-e795ad80-0cea-11e8-8438-602260f38b73.png)

![metadata page schemas](https://user-images.githubusercontent.com/2223916/36005125-ec257d12-0cea-11e8-87f8-ed7ddad5c30f.png)

**Comment by alasserr** [comment]
Hi,

where are we on this matter ? Last post was on Feb 9 .. This function would really be something great t have :)

Thanks for your work anyway,

Adrien

**Comment by cmregazzo** [comment]
Good afternoon. Is there any news on the subject?

I need to hide almost 1000 (thousand) tables.

Thank you.

Cristian Regazzo

**Comment by joboyx** [comment]
2 options that I can think of as workaround:

* Create a user specific for Metabase and don't give select/view access to those 1000+ tables. That way Metabase won't be able to introspect those tables thus it'll not show in UI
* Another option is to connect to Metabase' application DB. And via SQL update statement you can hide all those tables. There should be a hidden column there to tag it as hidden.

**Comment by d47id** [comment]
I was about to recommend trying Metabase at my organization, but ran across a need for this feature almost immediately while putting together a POC. 

I have to say, it's not very confidence-inspiring that this "quick and easy UX win" has been open for over 2 years. I'm probably going to look elsewhere at this point.

**Comment by joboyx** [comment]
You can manually do this by connecting and updating the Metabase application DB and setting specific tables to hidden.

There should be a `metabase_database` table.

**Comment by MrMauricioLeite** [comment]
just ran into this again...
need to hide 20 schemas all packed with tables...

Can't find a way to connect to the database and hide them via SQL. I'm using the Mac App for a quick and dirty POC but seems like a lot of effort. No idea when we can get something like this merged @metabaseops ? ( @mazameli and @camsaul for visibility)

**Comment by botchniaque** [comment]
Currently we have 1600+ tables, out of which we only want to expose couple of them - basically belonging to one particular schema. Having a _include_ rather than _exclude_ pattern would be better for us, since new unwanted schemas are likely to show up. That's also the reason why running `UPDATE SET visibility_type = 'hidden'` directly on a database is not a good solution, as we'd have to run it periodically. 

I imagine a usable solution as either 

- have a ___hidden___ flag on a schema level (Admin -> Data Model -> _select schema_)
- have a ___list of regular expressions___ of visible (or invisible) tables on a database level (Admin -> Databases -> _select database_)

**Comment by elise-lcdb** [comment]
Any updates on the issue?

**Comment by zcattacz** [comment]
What is most requested is definitely 2 features instead of 1:
- default setting for [ignoroe schema] or [hiding tables in schame] at database setup phase.
- ui widgets to corret current labor intesive setting.

It seem that the first issue is overlooked, despite the ticket is titled "as default"...

**Comment by flamber** [comment]
@zcattacz This issue is for the default value. There's #2086 for making the UI less labor intensive.
Related to point 2 of #6735

**Comment by Rodeoclash** [comment]
Another use case for this feature is when Metabase is working alongside Airbyte. Airbyte appears to be quite fond of creating temporary tables which all have the same naming pattern, but unique names.

The ability to hide tables by a regex on the name or to hide all tables by default and turn on only the ones I want would be two possible ways of solving this.

**Comment by flamber** [comment]
@Rodeoclash You're looking for #5281

**Comment by alexyarosh** [comment]
In v59, t[he Data Structure section of Data Studio](https://www.metabase.com/docs/latest/data-studio/data-structure) includes bulk actions capability for tables, including hiding tables in bulk.
